core/src/main/java/org/apache/brooklyn/entity/group/DynamicClusterImpl.java

```diff
@@ -151,12 +151,14 @@ public ZoneFailureDetector apply(final String input) {
      */
     protected final Object mutex = new Object[0];
 
-    private static final Function<Collection<Entity>, Entity> defaultRemovalStrategy = new Function<Collection<Entity>, Entity>() {
-        @Override public Entity apply(Collection<Entity> contenders) {
```

Comment:
Again on persistence. @aledsage is deleting this anonymous inner class safe? (seems a little silly to have to ask!)